src/object/cli_obj.c

```diff
@@ -3392,6 +3392,15 @@ obj_shard_list_comp_cb(struct shard_auxi_args *shard_auxi,
 
 	shard_arg = container_of(shard_auxi, struct shard_list_args, la_auxi);
 	if (obj_auxi->req_tgts.ort_grp_size == 1) {
+		if (obj_auxi->opc == DAOS_OBJ_RECX_RPC_ENUMERATE) {
```

Comment:
Following test should be added to properly handle use case where the function `daos_obj_list_recx()` is called with its parameter `recxs` equal to `NULL`. For example, in the _daos-serialize_ code, it is used to retrieve the size of the record.
```suggestion
		if (obj_auxi->opc == DAOS_OBJ_RECX_RPC_ENUMERATE &&
		    shard_arg->la_recxs != NULL) {
```

Reply:
ah, yes, thanks Cedric

Reply:
> ah, yes, thanks Cedric

FYI, with the little proposed patch, the ftest depoyment/iosysadmin.py is running like a charm :)
Thanks again for your help and this patch.